lib/internal/modules/helpers.js

```diff
@@ -318,23 +333,34 @@ function loadTypeScriptParser(parser) {
   } else {
     const amaro = require('internal/deps/amaro/dist/index');
     // Default option for Amaro is to perform Type Stripping only.
-    const defaultOptions = { __proto__: null, mode: 'strip-only' };
+    typeScriptParsingMode = getOptionValue('--experimental-enable-type-transform') ? 'transform' : 'strip-only';
+    sourceMapEnabled = getOptionValue('--enable-source-maps');
     // Curry the transformSync function with the default options.
-    typeScriptParser = (source) => amaro.transformSync(source, defaultOptions);
+    typeScriptParser = amaro.transformSync;
   }
   return typeScriptParser;
 }
 
 /**
+ * @typedef {object} TransformOutput
+ * @property {string} code - The compiled code.
+ * @property {string} [map] - The source maps (optional).
+ *
  * Performs type-stripping to TypeScript source code.
  * @param {string} source TypeScript code to parse.
- * @returns {string} JavaScript code.
+ * @param {string} filename The filename of the source code.
+ * @returns {TransformOutput} The stripped TypeScript code.
  */
-function tsParse(source) {
+function stripTypes(source, filename) {
   // TODO(@marco-ippolito) Checking empty string or non string input should be handled in Amaro.
   if (!source || typeof source !== 'string') { return ''; }
   const parse = loadTypeScriptParser();
-  const { code } = parse(source);
+  const options = { __proto__: null, mode: typeScriptParsingMode, sourceMap: sourceMapEnabled, filename };
+  const { code, map } = parse(source, options);
+  if (map) {
+    const base64SourceMap = Buffer.from(map).toString('base64');
```

Comment:
Can you add a TODO to here?

When Buffer.transcode supports utf8 to base64 transformation, we should change this line.